lib/route/link/macvlan.c

```diff
@@ -277,6 +374,55 @@ uint32_t rtnl_link_macvlan_get_mode(struct rtnl_link *link)
 }
 
 /**
+ * Set MACVLAN MACMODE
+ * @arg link		Link object
+ * @arg mode		MACVLAN mac list modification mode
+ *
+ * Only for macvlan SOURCE mode.
+ *
+ * @return 0 on success or a negative error code
+ */
+int rtnl_link_macvlan_set_macmode(struct rtnl_link *link, uint32_t macmode)
+{
+	struct macvlan_info *mvi = link->l_info;
+
+	IS_MACVLAN_LINK_ASSERT(link);
+
+	if (!(mvi->mvi_mask & MACVLAN_HAS_MODE) ||
+	    (mvi->mvi_mode != MACVLAN_MODE_SOURCE)) {
+		APPBUG("SET_MACMODE is supported with macvlan source mode.");
+		return -NLE_OPNOTSUPP;
+	}
+
+	mvi->mvi_macmode = macmode;
+
+	return 0;
+}
+
+/**
+ * Get MACVLAN MACMODE
+ * @arg link		Link object
+ *
+ * Only for SOURCE mode.
+ *
+ * @return MACVLAN mac list modification mode, 0 if not set or a negative error code.
+ */
+int rtnl_link_macvlan_get_macmode(struct rtnl_link *link)
```

Comment:
can we change the signature to

int rtnl_link_macvlan_get_macmode(struct rtnl_link *link, uint32_t *out_macmode);

libnl is unfortunately not consistent about this, but I think getters should return an error-code or success only. The actual return value should be passed separately. Especially, as uint32_t doesn't fit entirely in the "int" return value.